### ISSUE TITLE
Private package config not working with bit.dev 

### ISSUE THREAD

**Issue by lxnre-codes** [oven-sh/bun#4816]
### What version of Bun is running?

1.0.0

### What platform is your computer?

Darwin 22.6.0 x86_64 i386

### What steps can reproduce the bug?

doesn't work with bunfig.toml
```toml
# bunfig.toml 

[install.scopes]
# does not work
@myorg = { token = "my_token", url = "https://node.bit.cloud" }

# this works 
@myorg2 = { token = "my_token", url = "https://npm.pkg.github.com" }
```

This works with .npmrc 

```npmrc
@myorg:registry=https://node.bit.cloud
//node.bitsrc.io/:_authToken=my_token
//node.bit.cloud/:_authToken=my_token

@myorg2:registry=https://registry.npmjs.org/
@myorg2:registry=https://npm.pkg.github.com/
//npm.pkg.github.com/:_authToken=my_token
```

### What is the expected behavior?

Private repository config should authenticate with bit cloud

### What do you see instead?

Error 404. Failed to resolve package 
<img width="348" alt="Screenshot 2023-09-10 at 9 45 08 AM" src="https://github.com/oven-sh/bun/assets/74317911/1d29bedc-8c02-4b92-a5ee-4c7e6ca45e87">

**Comment by inimicus** [comment]
@0x-buidl Is this in your project's root directory, e.g. alongside `package.json`?

I had a similar issue that was resolved by moving, undesirably, `.bunfig.toml` to `~/`.

If so, I believe the issue is related to #4901 .

**Comment by lxnre-codes** [comment]
This doesn't work for `bit.dev` either

**Comment by lxnre-codes** [comment]
fixed with #5906
